File: briar-android/src/main/java/org/briarproject/briar/android/controller/BriarController.java
package org.briarproject.briar.android.controller;

import org.briarproject.bramble.api.nullsafety.NotNullByDefault;
import org.briarproject.bramble.api.system.Wakeful;
import org.briarproject.briar.android.controller.handler.ResultHandler;

@NotNullByDefault
public interface BriarController extends ActivityLifecycleController {

	void startAndBindService();

	boolean accountSignedIn();

	/**
	 * Returns true via the handler when the app has dozed
	 * without being white-listed.
	 */
	void hasDozed(ResultHandler<Boolean> handler);

	void doNotAskAgainForDozeWhiteListing();

	@Wakeful
	void signOut(ResultHandler<Void> handler, boolean deleteAccount);

	void deleteAccount();

}

File: briar-android/src/main/kotlin/org/briarproject/briar/android/controller/BriarController.kt
package org.briarproject.briar.android.controller

import org.briarproject.bramble.api.system.Wakeful
import org.briarproject.briar.android.controller.handler.ResultHandler

interface BriarController : ActivityLifecycleController {

	fun startAndBindService()

	fun accountSignedIn(): Boolean

	fun isTelegramConnectorReady(): Boolean

	fun getTelegramLinkedIdentity(handler: ResultHandler<String>)

	/**
	 * Returns true via handler when app has dozed
	 * without being white-listed.
	 */
	fun hasDozed(handler: ResultHandler<Boolean>)

	fun doNotAskAgainForDozeWhiteListing()

	@Wakeful
	fun signOut(handler: ResultHandler<Void>, deleteAccount: Boolean)

	fun deleteAccount()

}
